### PR TITLE
Implement hpegl_vmaas_router_route resource

### PR DESCRIPTION
**Pull request by tshihad**

### CODE REVIEW

**Review by gandharvas — DISMISSED** [review]

File: internal/cmp/router_helper.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+) (bool, error) {
+	*isDeprecated = false
+	// Get the router, if the router not exists, return warning
+	router, err := rClient.GetSpecificRouter(ctx, routerID)
+	if err != nil {
+		return false, err
+	}
+	// if router not found set is_deprecated flag=true
+	if router.ID == 0 {
+		log.Printf("[ERROR] Router with %d id is not found on NAT plan", routerID)
```
Comment (manjunath-batakurki) [inline internal/cmp/router_helper.go:30]: This can be used for other resources under Router.

**Review by gandharvas — APPROVED** [review]

File: internal/cmp/router_route.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		return nil
+	}
+
+	resp, err := r.rClient.DeleteRouterRoute(ctx, tfRoute.RouterID, tfRoute.ID)
+	if err != nil {
+		return err
+	}
+	if !resp.Success {
+		return fmt.Errorf("got success = 'false' while deleting NAT rule")
+	}
```
Comment (manjunath-batakurki) [inline internal/cmp/router_route.go:95]: Route

File: internal/cmp/router_route.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+func (r *routerRoute) Create(ctx context.Context, d *utils.Data, meta interface{}) error {
+	setMeta(meta, r.rClient.Client)
+	var tfRoute models.RouterRouteBody
+	err := tftags.Get(d, &tfRoute)
+	if err != nil {
+		return err
+	}
+	natRes, err := r.rClient.CreateRouterRoute(ctx, tfRoute.RouterID,
```
Comment (manjunath-batakurki) [inline internal/cmp/router_route.go:54]: routeRes

File: internal/cmp/router_route.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+		models.CreateRouterRoute{
+			NetworkRoute: tfRoute,
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	if !natRes.Success {
```
Comment (manjunath-batakurki) [inline internal/cmp/router_route.go:63]: routeRes

File: internal/cmp/router_route.go [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+		return fmt.Errorf(successErr, "creating route for the router")
+	}
+	tfRoute.ID = natRes.ID
```
Comment (manjunath-batakurki) [inline internal/cmp/router_route.go:66]: routeRes

**Review by manjunath-batakurki — APPROVED** [review]
LGTM

**Review by manjunath-batakurki — APPROVED** [review]
LGTM

**Review by gandharvas — APPROVED** [review]

**Review by manjunath-batakurki — APPROVED** [review]
LGTM

**Pull request merged**